modules/extensions/src/collide/shader-module.ts

```diff
@@ -0,0 +1,89 @@
+import {project} from '@deck.gl/core';
+import type {_ShaderModule as ShaderModule} from '@deck.gl/core';
+import type {Texture2D} from '@luma.gl/webgl';
+
+const vs = `
+#ifdef NON_INSTANCED_MODEL
+attribute float collidePriorities;
+#else
+attribute float instanceCollidePriorities;
+#endif
+
+uniform sampler2D collide_texture;
+uniform bool collide_sort;
+uniform bool collide_enabled;
+
+vec2 collide_getCoords(vec4 position) {
+  vec4 collide_clipspace = project_common_position_to_clipspace(position);
+  return (1.0 + collide_clipspace.xy / collide_clipspace.w) / 2.0;
+}
+
+float collide_match(vec2 tex, vec3 pickingColor) {
+  vec4 collide_pickingColor = texture2D(collide_texture, tex);
+  float delta = dot(abs(collide_pickingColor.rgb - pickingColor), vec3(1.0));
+  float e = 0.001;
+  return step(delta, e);
+}
+
+float collide_isVisible(vec2 texCoords, vec3 pickingColor) {
+  if (!collide_enabled) {
+    return 1.0;
+  }
+
+  // Visibility test, sample area of 5x5 pixels in order to fade in/out.
+  // Due to the locality, the lookups will be cached
+  // This reduces the flicker present when objects are shown/hidden
+  const int N = 2;
+  float accumulator = 0.0;
+  vec2 step = vec2(1.0 / project_uViewportSize);
+
+  const float floatN = float(N);
+  vec2 delta = -floatN * step;
```

Comment:
How's the perf here?

Reply:
Good, I don't see any difference between doing a single lookup and sampling over an area and I think having the fading makes a big difference to the UX